src/main.ts

```diff
@@ -261,8 +261,26 @@ app.on('ready', () => {
     autoUpdater.checkForUpdates().catch(reportRejection)
   }, fifteenMinutes)
 
+  autoUpdater.on('error', (error) => {
+    console.error('updater-error', error)
+    mainWindow?.webContents.send('updater-error', error)
+  })
+
   autoUpdater.on('update-available', (info) => {
     console.log('update-available', info)
+    autoUpdater.prependOnceListener('download-progress', (progress) => {
+      // For now, we'll send the version to the loading toast start.
+      console.log('update-download-start', {
+        version: info.version,
+      })
+      mainWindow?.webContents.send('update-download-start', progress)
+    })
+
+    autoUpdater.on('download-progress', (progress) => {
+      // TODO: in a future PR (https://github.com/KittyCAD/modeling-app/issues/3994)
+      // send this data to mainWindow to show a progress bar for the download.
+      console.log('download-progress', progress)
+    })
```

Comment:
@franknoirot Shouldn't these live outside of the update-available callback?

Reply:
Yeah you're probably correct. The problem I had is that I want access to the version number in my loading toast, and it didn't seem like the `download-progress` provides that version number. Do you know how I could slot that data in?

Reply:
@franknoirot Tested that quickly https://github.com/KittyCAD/modeling-app/compare/franknoirot/update-download-progress...refs/heads/pierremtb/update-download-progress-flatten

You can grab the updater-test artifacts at https://github.com/KittyCAD/modeling-app/actions/runs/11142991008. Windows should be a be representation because the diff download stuff isn't working well I think and the download is much longer. In my testing I was able to see the loading toast (with version undefined of course) and then the update one.

I believe we need to either find a way to store the version outside of the hooks or change the language to something version agnostic

Reply:
Hold on I'm a little confused now. Clearing the cache gets me the expected behaviour from builds from your branch (this PR) too

Reply:
Well I think you're right here, flattening these is the right way to do things. I'll incorporate this change and simplify the loading toast to just not need the version number, that's the simplest way to ship this.

Reply:
I merged in your branch!

Reply:
> Hold on I'm a little confused now. Clearing the cache gets me the expected behaviour from builds from your branch (this PR) too (I'm seeing undefined for the version though)

I'm not sure what's up with that, but if this is looking good to you I'm down to get it in, then later we can do the progress bar across a custom toast wrapper.

Reply:
>Also, note that if you've had the 0.255.255 file downloaded before by running an updater-test on your machine, that file is cached and the download is instant!!

This makes sense hah I've gotta clear those files out

Reply:
For sure, I think this is great already, thanks!